src/ode_solver/ode_solver_factory.cpp

```diff
@@ -125,6 +98,132 @@ void ODESolverFactory<dim,real,MeshType>::display_error_ode_solver_factory(Param
     std::abort();
 }
 
+template <int dim, typename real, typename MeshType>
+std::shared_ptr<ODESolverBase<dim,real,MeshType>> ODESolverFactory<dim,real,MeshType>::create_RungeKuttaODESolver(std::shared_ptr< DGBase<dim,real,MeshType> > dg_input)
+{
+    dealii::ConditionalOStream pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD)==0);
+
+    std::shared_ptr<RKTableauBase<dim,real,MeshType>> rk_tableau = create_RKTableau(dg_input);
+   
+
+    const int n_rk_stages = dg_input->all_parameters->ode_solver_param.n_rk_stages;
+    using ODEEnum = Parameters::ODESolverParam::ODESolverEnum;
+    ODEEnum ode_solver_type = dg_input->all_parameters->ode_solver_param.ode_solver_type;
+    if (ode_solver_type == ODEEnum::runge_kutta_solver) {
+
+        pcout << "Creating Runge Kutta ODE Solver";
+        if (n_rk_stages == 1){
+            pcout << " with 1 stage..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,1,MeshType>>(dg_input,rk_tableau);
+        }
+        if (n_rk_stages == 2){
+            pcout << " with 2 stages..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,2,MeshType>>(dg_input,rk_tableau);
+        }
+        if (n_rk_stages == 3){
+            pcout << " with 3 stages..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,3,MeshType>>(dg_input,rk_tableau);
+        }
+        if (n_rk_stages == 4){
+            pcout << " with 4 stages..." << std::endl;
+            return std::make_shared<RungeKuttaODESolver<dim,real,4,MeshType>>(dg_input,rk_tableau);
+        }
```

Comment:
these can all be combined making use of the `n_rk_stages` variable within `if (n_rk_stages <= 4)`

Reply:
Unfortunately, it seems like the hard-coded structure I've used is necessary because `n_rk_stages` is not known at compile-time. I've restructured the `pcout` statements to make the structure a little cleaner, and added a comment justifying the hard-coding.